src/odbc/src/app/application_data_buffer.cpp

```diff
@@ -278,15 +279,14 @@ ConversionResult::Type ApplicationDataBuffer::PutStrToStrBuffer(
   }
 
   SqlUlen currentCellOffset = cellOffset >= 0 ? cellOffset : 0;
-  // Since cellOffset is in bytes, an index needs to be calculated.
-  SqlUlen inCharIndex = currentCellOffset / inCharSize;
 
-  if (inCharIndex >= value.length()) {
-    if (resLenPtr) {
-      *resLenPtr = SQL_NO_TOTAL;
-    }
+  // If all data has already been read, return AI_NO_DATA
+  if ((bytesRequired - ((currentCellOffset / inCharSize) * outCharSize)) <= 0) {
```

Comment:
This check should be refactored because both bytesRequired and currentCellOffset are unsigned integers so the result of subtraction can't be < 0.